src/coreclr/tools/Common/Internal/NativeFormat/NativeFormatWriter.cs

```diff
@@ -2044,12 +2052,20 @@ internal override void Save(NativeWriter writer)
 
         public override bool Equals(object obj)
         {
-            throw new NotImplementedException();
+            // TODO: Support value equality instead of just referential equality
+            return ReferenceEquals(this, obj);
         }
 
         public override int GetHashCode()
         {
-            throw new NotImplementedException();
+            int hashCode = 13;
+            foreach (var entry in _Entries)
+            {
+                int value = (int)entry.Hashcode * 0x5498341 + 0x832424;
+                hashCode = hashCode * 31 + value;
+            }
```

Comment:
Please document this algorithm, or preferrably can we use `HashCode`?

Reply:
I think we can use `HashCode` here. I'll try.